includes/amp-helper-functions.php

```diff
@@ -603,10 +603,15 @@ function amp_get_slug() {
  *
  * @since 1.0
  *
- * @return string Current URL.
+ * @return string|null Current URL, or null if in a WP CLI context and the home URL could not be parsed.
  */
 function amp_get_current_url() {
 	$parsed_url = wp_parse_url( home_url() );
+
+	if ( defined( 'WP_CLI' ) && WP_CLI && ! is_array( $parsed_url ) ) {
+		return null;
+	}
```

Comment:
How about faking a URL on the `localhost` domain?

Reply:
Yes, that makes sense to me.

Reply:
Like:

```suggestion
	if ( ! is_array( $parsed_url ) ) {
		$parsed_url = wp_parse_url( 'https://localhost' );
	}
```

Then we don't need the condition below where `$parsed_url` is set to `[]`.

Reply:
Or rather:

```suggestion
	if ( ! is_array( $parsed_url ) ) {
		$parsed_url = [
			'host' => 'localhost',
		];
	}
```

Reply:
Addressed in https://github.com/ampproject/amp-wp/pull/5219/commits/34c6060bf0b05df70ab028a1d68daf55d151c1c3.